hack/machineset.sh

```diff
@@ -173,6 +173,74 @@ $(get_spec $infraID $az $provider)
 EOF
 }
 
+# get_vsphere_ms creates a MachineSet for vSphere Cloud Provider
+get_vsphere_ms() {
+
+  if [ "$#" -lt 2 ]; then
+    error-exit incorrect parameter count for get_vsphere_ms $#
+  fi
+
+  local infraID=$1
+  local provider=$2
+
+  # set golden image template name
+  # TODO: read from parameter
+  template="windows-golden-images/windows-server-2004-template"
+
+  # get vsphere providerSpec for any worker machine
+  # TODO: filter machines in running phase
+  providerSpec=$(oc get machines \
+                -n openshift-machine-api \
+                -l machine.openshift.io/cluster-api-machine-role=worker \
+                -o jsonpath="{.items[0].spec.providerSpec.value}" \
+  ) || {
+    error-exit "error getting providerSpec for ${provider} cluster ${infraID}"
+  }
+  if [ -z "$providerSpec" ]; then
+    error-exit "cannot find providerSpec for ${provider} cluster ${infraID}"
+  fi
+  # get credentialsSecret
+  credentialsSecret=$(echo "${providerSpec}" | jq -r '.credentialsSecret.name')
+  # get network name TODO: review when devices > 1
+  networkName=$(echo "${providerSpec}" | jq -r '.network.devices[0].networkName')
+  # get workspace specs
+  datacenter=$(echo "${providerSpec}" | jq -r '.workspace.datacenter')
+  datastore=$(echo "${providerSpec}" | jq -r '.workspace.datastore')
+  folder=$(echo "${providerSpec}" | jq -r '.workspace.folder')
+  resourcePool=$(echo "${providerSpec}" | jq -r '.workspace.resourcePool')
+  server=$(echo "${providerSpec}" | jq -r '.workspace.server')
+  # build machineset
+  cat <<EOF
+$(get_spec $infraID "" $provider)
+      providerSpec:
+        value:
+          apiVersion: vsphereprovider.openshift.io/v1beta1
+          credentialsSecret:
+            name: ${credentialsSecret}
+          diskGiB: 128
+          kind: VSphereMachineProviderSpec
+          memoryMiB: 8192
+          metadata:
+            creationTimestamp: null
+          network:
+            devices:
+            - networkName: ${networkName}
+          numCPUs: 4
+          numCoresPerSocket: 1
+          snapshot: ""
+          template: ${template}
+          # TODO: remove; WMCO does not use userDataSecret in vSphere platform
+          userDataSecret:
+            name: windows-user-data
```

Comment:
Planning to remove `userDataSecret` since not used in vSphere platform.

Thoughts?

Reply:
Ended up removing the `userDataSecret` section for vSphere machineSet; 
a) YAGNI, the pre-reqs are met in the golden image, and
b) allows creating a machineSet before WMCO is deployed (BYOH use case). Otherwise, we depends on WMCO to create `windows-user-data`.